src/commands/deploy.js

```diff
@@ -472,6 +482,11 @@ DeployCommand.flags = {
     char: 'f',
     description: 'Specify a functions folder to deploy',
   }),
+  edgeHandlers: flags.string({
```

Comment:
I think users should still be able to override it via a CLI flag (like we have `--dir` flag and `--functions` flag).
WDYT?

Reply:
I think I got confused in my original comment because `edgeHandlers` is the non-configurable destination directory (`.netlify/edge-handlers/`), not the configurable source directory (`./edge-handlers`).

Isn't `--functions` for the Functions source directory, as opposed to the destination directory? For example, `./functions` (where user is writing source files) as opposed to `.netlify/functions/` (where they are output). The difference might be important because source directories are exposed to users and configurable (in `netlify.toml`) whereas destination directories are not exposed to users (they are implementation details) and non-configurable. From that angle, I am not sure whether we want to make it configurable?

Please let me know if I mixed up some directories :)

Reply:
Very good point.
That is the value for the post build bundle location of edge handlers and not the source directory.

In general I think we should have good defaults, but still allow overriding them - this is unrelated to the specific issue at hand and the semantics of the directory.

After going over https://github.com/netlify/build/pull/1828#discussion_r491675488 I think we could have 2 flags:
1. `edgeHandlers` which is the source of edge handlers (assuming this is configurable). The default should come from Netlify config, but overridable via the CLI flag. This is probably only useful for the build command and not the deploy command.
2. `edgeHandlersBundleLocation` (or something shorter) which is the destination directory. This should also default to a value provided by Netlify config, but overridable via the CLI flag. This for example can allow an advanced use case for users bundling their own edge handlers into a specific directory.

The latter could remain hidden as for a common use case it should not be used.

WDYT?

Reply:
Following up after reading https://github.com/netlify/build/issues/1822 as we might be going with a hardcoded solution pending that issue. We need to wait for that discussion to resolve.

Reply:
For `edgeHandlersBundleLocation`, I think we should be consistent with whatever solution is preferred in https://github.com/netlify/build/issues/1822. If we think that the destination directory should not be exposed to users as a plugin `constants`, it seems to me this decision would imply not providing a CLI flag. On the other hand, if we do want to expose it, creating a CLI flag would make sense IMHO.

> edgeHandlers which is the source of edge handlers (assuming this is configurable). The default should come from Netlify config, but overridable via the CLI flag. This is probably only useful for the build command and not the deploy command.

Yes, this should only be useful for the `netlify build` command since the deploy command is only concerned with the destination directory. 

Users should be able to configure this using their `netlify.toml` (https://github.com/netlify/build/pull/1828). At the moment, we don't provide any CLI flags in `netlify build` for overriding build settings. Instead, we only take them from the `netlify.toml` and the UI build settings. The problem I could see with adding such CLI flags is that their number could quickly grow: if we add `--edge-handlers`, we might also want to add `--functions`, `--publish`, `--env`, `--base`, etc.

Note that there is an `--inlineConfig` CLI flag for `@netlify/config` which could work perfectly for those flags. It is a way to provide an "inline" `netlify.toml` object that has the highest priority. E.g. `--inlineConfig.build.edge_handlers` could be used in this case.

I think this might be worth opening a separate issue since this might be a more general question: should we support overriding build settings with `netlify build` CLI flags?

Reply:
Everything you said makes sense. I won't add the `edge-handlers` flag in this PR, and for `edgeHandlersBundleLocation` we'll need to wait.

> E.g. `--inlineConfig.build.edge_handlers` could be used in this case. This allows adding all those flags in the CLI without having to add corresponding flags in `@netlify/config`.

Cool, to clarify - I wasn't suggesting we add those to `@netlify/config`, just to be able to override those using the CLI.
These flags are useful when running the CLI in CI, where you want to override specific settings (and don't want to start modifying files for the sake of it).
Using `inlineConfig` should work very well for that.

> should we support overriding build settings with `netlify build` CLI flags?
Yes, this a good topic for discussion - I'll open a new issue for it

Reply:
Should we merge this PR even if the `edgeHandlersBundleLocation` discussion is still pending?

Reply:
Ok, I'll remove the flag, use a hardcoded version, then merge.
We can add the flag later on

Reply:
Done in https://github.com/netlify/cli/pull/1244/commits/f057acf91904a5edf3ab84e884b89c35721a430c